sdk/search/Azure.Search.Documents/tests/Samples/FieldBuilder/FieldBuilder.cs

```diff
@@ -0,0 +1,451 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Reflection;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Azure.Search.Documents.Indexes.Models;
+#if EXPERIMENTAL_SPATIAL
+using Azure.Core.Spatial;
+#endif
+
+namespace Azure.Search.Documents.Tests.Samples
+{
+    /// <summary>
+    /// Builds field definitions for a search index by reflecting over a user-defined model type.
+    /// </summary>
+    /// <remarks>
+    /// <para>
+    /// This <see cref="FieldBuilder"/> was ported from the Microsoft.Azure.Search.Service package
+    /// to make migrating from using Microsoft.Azure.Search to Azure.Search.Documents easier.
+    /// It also uses System.Text.Json instead of Newtonsoft.Json (JSON.NET).
+    /// </para>
+    /// <para>
+    /// This is only a sample you can include in your code and future implementations may change
+    /// to follow modern guidelines and design principles.
+    /// </para>
+    /// </remarks>
+    public static class FieldBuilder
+    {
+        private static readonly IReadOnlyDictionary<Type, SearchFieldDataType> s_primitiveTypeMap =
+            new ReadOnlyDictionary<Type, SearchFieldDataType>(
+                new Dictionary<Type, SearchFieldDataType>()
+                {
+                    [typeof(string)] = SearchFieldDataType.String,
+                    [typeof(int)] = SearchFieldDataType.Int32,
```

Comment:
Should we map other values like short/byte/etc. to int?  Same for float to double?

Reply:
I had considered that, but this was just a port of track 1 FieldBuilder. I wanted to keep it the same as much as possible and not invest too much time in what is a sample, and even states that the final API will (may have said "may") be different. If you feel strongly, please reopen.